apps/api/src/app/subscribers-v2/usecases/create-subscriber/create-subscriber.usecase.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+import { SubscriberEntity, SubscriberRepository } from '@novu/dal';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { SubscriberResponseDto } from '../../../subscribers/dtos';
+import { mapSubscriberEntityToDto } from '../list-subscribers/map-subscriber-entity-to.dto';
+import { CreateSubscriberCommand } from './create-subscriber.command';
+
+@Injectable()
+export class CreateSubscriber {
+  constructor(private subscriberRepository: SubscriberRepository) {}
+
+  async execute(command: CreateSubscriberCommand): Promise<SubscriberResponseDto> {
+    const existingSubscriber = await this.subscriberRepository.findOne({
+      subscriberId: command.createSubscriberRequestDto.subscriberId,
+      _environmentId: command.environmentId,
+      _organizationId: command.organizationId,
+    });
+
+    if (existingSubscriber) {
+      throw new ApiException(`Subscriber: ${command.createSubscriberRequestDto.subscriberId} already exists`);
```

Comment:
Shoudl we throw an ApiException or a Validation exception?

Reply:
I think if a user post the API with an existing subscriber id, it can be treated as a Bad Request 400 code

Reply:
Sound good.